lib/Services/GrantService.php

```diff
@@ -28,27 +28,49 @@ class GrantService implements GrantRoutes
 
     private PendingGrantTransformer $pendingTransformer;
 
+    private string $clientUrl = '';
+
     /**
      * GrantService constructor.
      *
      * @param  GrantValidator|null  $validator
      */
     public function __construct(
         ApiClient $apiClient,
+        string $clientUrl = '',
         ?GrantTransformer $grantTransformer = null,
         ?PendingGrantTransformer $pendingTransformer = null
     ) {
         $this->apiClient = $apiClient;
+        $this->clientUrl = $clientUrl;
         $this->validator = new GrantValidator;
         $this->grantTransformer = $grantTransformer ?? new GrantTransformer;
-        $this->pendingTransformer = $pendingTransformer ?? new PendingGrantTransformer;
+        $this->pendingTransformer
+            = $pendingTransformer ?? new PendingGrantTransformer;
+    }
+
+    public function addClientToStruct(array $grantStructure): array
+    {
+        $grantStructure['client'] = $this->clientUrl;
+
+        return $grantStructure;
     }
 
+    /**
+     * Requests a new grant or pending grant.
+     *
+     * @param  array  $requestParams  Parameters for the request, including 'url'.
+     * @param  array  $grantRequest  The grant request data.
+     *
+     * @throws \InvalidArgumentException
+     * @throws \Exception
+     */
     public function request(array $requestParams, array $grantRequest): Grant|PendingGrant
     {
         if (! isset($requestParams['url'])) {
             throw new \InvalidArgumentException('Missing required data');
         }
+        $grantRequest = $this->addClientToStruct($grantRequest);
```

Comment:
Related to above comment about accepting a reference (if applicable):
```suggestion
        $this->injectClientUrl($grantRequest);
```